src/main/java/co/rsk/federate/config/PowHSMBookkeepingConfig.java

```diff
@@ -1,87 +1,86 @@
 package co.rsk.federate.config;
 
-/**
- * Represents the configuration for a signer.
- * Mainly has an identifier for the signer, the
- * type of signer and additional configuration options.
- *
- * @author Pamela Gonzalez
- */
-
 import co.rsk.bitcoinj.core.NetworkParameters;
+import java.math.BigInteger;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.math.BigInteger;
-
 public class PowHSMBookkeepingConfig {
 
     private static final Logger logger = LoggerFactory.getLogger(PowHSMBookkeepingConfig.class);
-    public static final BigInteger DIFFICULTY_CAP_MAINNET = new BigInteger("7000000000000000000000");
-    public static final BigInteger DIFFICULTY_CAP_TESTNET = BigInteger.valueOf(1000000000000000L);
-    public static final BigInteger DIFFICULTY_CAP_REGTEST = BigInteger.valueOf(20);
-    private final BigInteger difficultyCap;
-    private BigInteger difficultyTarget = BigInteger.valueOf(3);
-    private int maxAmountBlockHeaders = 7;
-    private long informerInterval = 2_000;
-    private boolean stopBookkeepingScheduler = false;
-    private int maxChunkSizeToHsm = 10;
 
-    public PowHSMBookkeepingConfig(SignerConfig signerConfig, String networkParameter) {
-        if (signerConfig.getConfig().hasPath("bookkeeping.difficultyTarget")) {
-            this.difficultyTarget = new BigInteger(signerConfig.getConfig().getString("bookkeeping.difficultyTarget"));
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxAmountBlockHeaders")) {
-            this.maxAmountBlockHeaders = signerConfig.getConfig().getInt("bookkeeping.maxAmountBlockHeaders");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.informerInterval")) {
-            this.informerInterval = signerConfig.getConfig().getLong("bookkeeping.informerInterval");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.stopBookkeepingScheduler")) {
-            this.stopBookkeepingScheduler = signerConfig.getConfig().getBoolean("bookkeeping.stopBookkeepingScheduler");
-        }
-        if (signerConfig.getConfig().hasPath("bookkeeping.maxChunkSizeToHsm")) {
-            this.maxChunkSizeToHsm = signerConfig.getConfig().getInt("bookkeeping.maxChunkSizeToHsm");
+    public enum NetworkDifficultyCap {
+        MAINNET(new BigInteger("7000000000000000000000")),
+        TESTNET(BigInteger.valueOf(1000000000000000L)),
+        REGTEST(BigInteger.valueOf(20L));
+
+        private final BigInteger difficultyCap;
+
+        NetworkDifficultyCap(BigInteger difficultyCap) {
+            this.difficultyCap = difficultyCap;
```

Comment:
would it make sense to have this enum in a separated class?

Reply:
Since it is only being consumed inside the scope of this class it can be left here for now. If it starts being consumed by two or more classes then we can extract it.